search/search_param_types.go

```diff
@@ -289,6 +296,29 @@ func (q *Query) UsesPipeline() bool {
 	return q.UsesIncludes() || q.UsesRevIncludes() || q.UsesChainedSearch() || q.UsesReverseChainedSearch()
 }
 
+// SupportsPaging returns true if the query results can be paginated, false if not.
+// In practice, most queries can be paginated, but queries for $everything or _summary cannot.
+func (q *Query) SupportsPaging() bool {
+	options := q.Options()
+
+	// not for $everything. $everything is defined as _id=<id>&_include=*&_revinclude=*
+	if q.isDollarEverything() {
+		return false
+	}
+
+	// not for _summary=count
+	if options.Summary == "count" {
+		return false
+	}
+
+	return true
+}
+
+func (q *Query) isDollarEverything() bool {
+	de := regexp.MustCompile("_id=[0-9a-f]{24}&_include=\\*&_revinclude=\\*")
+	return de.MatchString(q.Query)
```

Comment:
This is fine, but you probably could have gotten away with something like `"_id=[^,&]+&_include=\\*&_revinclude=\\*"` (which would have allowed your test ids to remain the same).